scalding-core/src/main/scala/com/twitter/scalding/typed/TypedPipe.scala

```diff
@@ -137,6 +137,52 @@ object TypedPipe extends Serializable {
     }
   }
 
+  /**
+   * Which TypedPipes does the given TypedPipe directly depend on
+   * this must form a DAG by construction. This is useful for
+   * backend implementations.
+   */
+  def dependencies(of: TypedPipe[Any]): List[TypedPipe[Any]] =
```

Comment:
to confirm, you're using Any here cause we don't know of the types of the dependency TypedPipes right?